src/js/popup/popup.js

```diff
@@ -272,12 +272,15 @@ async function showRelayPanel(tipPanelToShow) {
   // const isBundleAvailableInCountry = (await browser.storage.local.get("bundlePlans")).bundlePlans.BUNDLE_PLANS.available_in_country;
   const isPhoneAvailableInCountry = (await browser.storage.local.get("phonePlans")).phonePlans.PHONE_PLANS.available_in_country;
   
-  const phoneMaskingAvailable =    await checkWaffleFlag("phones") && isPhoneAvailableInCountry;
+  // If user has a phone plan, don't show the phone masking promo
+  const hasPhone = (await browser.storage.local.get("has_phone")).has_phone;
```

Comment:
Question: Are there any additional data catches/failsafes if this check doesn't happen? _(Re: what happened last release)_

Reply:
Shouldn't the earlier fix ensure that profile data becomes updated upon a version release?

Reply:
I also tested this with a new Nightly profile and setting up `xpinstall.signatures.required` to false etc. It _should_ just revert back to the original state if the fetch isn't successful. Let's make sure QA touches this first before submitting too.